### PR TITLE
MM-49845: fixes acknowledgements in getpostsince

### PR DESCRIPTION
**Pull request by koox00**
#### Summary

GetPostsSince returns posts updated after a user requested date. If a user acknowledges a post the post's update at won't change thus not getting that post returned from GetPostsSince. This is troublesome particularly for the mobile client since it relies a lot to GetPostsSince for keeping the data up to date.

This commit updates the post's update_at when a user acknowledges/un-acknowledges a post fixing this way the issue above.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-49845

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/post_acknowledgements_store.go [hunk 1]
```diff
@@ -86,11 +108,21 @@ func (s *SqlPostAcknowledgementStore) Delete(ack *model.PostAcknowledgement) err
...
 		})
 
-	_, err := s.GetMasterX().ExecBuilder(query)
+	_, err = transaction.ExecBuilder(query)
 	if err != nil {
 		return err
 	}
 
+	err = updatePost(transaction, ack.PostId)
+
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:117]: style nit:
```suggestion
```

File: store/sqlstore/post_acknowledgements_store.go [hunk 2]
```diff
@@ -86,11 +108,21 @@ func (s *SqlPostAcknowledgementStore) Delete(ack *model.PostAcknowledgement) err
...
+	if err != nil {
+		return err
+	}
+
+	if err := transaction.Commit(); err != nil {
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:122]: style nit:
```suggestion
	err = transaction.Commit()
	if err != nil {
```

File: store/sqlstore/post_acknowledgements_store.go [hunk 3]
```diff
@@ -69,15 +75,31 @@ func (s *SqlPostAcknowledgementStore) Save(postID, userID string, acknowledgedAt
...
 		return nil, err
 	}
 
+	err = updatePost(transaction, acknowledgement.PostId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := transaction.Commit(); err != nil {
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:89]: style nit:
```suggestion
	err = transaction.Commit()
	if err != nil {
```

File: store/sqlstore/post_acknowledgements_store.go [hunk 4]
```diff
@@ -69,15 +75,31 @@ func (s *SqlPostAcknowledgementStore) Save(postID, userID string, acknowledgedAt
...
 	}
 
-	_, err := s.GetMasterX().ExecBuilder(query)
+	_, err = transaction.ExecBuilder(query)
 	if err != nil {
 		return nil, err
 	}
 
+	err = updatePost(transaction, acknowledgement.PostId)
+
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:84]: style nit:
```suggestion
```

File: store/sqlstore/post_acknowledgements_store.go [hunk 5]
```diff
@@ -69,15 +75,31 @@ func (s *SqlPostAcknowledgementStore) Save(postID, userID string, acknowledgedAt
...
 	return acknowledgement, nil
 }
 
 func (s *SqlPostAcknowledgementStore) Delete(ack *model.PostAcknowledgement) error {
+	transaction, err := s.GetMasterX().Beginx()
+	if err != nil {
+		return errors.Wrap(err, "begin_transaction")
+	}
+
+	defer finalizeTransactionX(transaction, &err)
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:102]: style nit:
```suggestion
	defer finalizeTransactionX(transaction, &err)
	
```

File: store/sqlstore/post_acknowledgements_store.go [hunk 6]
```diff
@@ -86,11 +108,21 @@ func (s *SqlPostAcknowledgementStore) Delete(ack *model.PostAcknowledgement) err
 			sq.Eq{"UserId": ack.UserId},
 		})
 
-	_, err := s.GetMasterX().ExecBuilder(query)
+	_, err = transaction.ExecBuilder(query)
 	if err != nil {
 		return err
 	}
 
+	err = updatePost(transaction, ack.PostId)
```
Comment (lieut-data) [inline store/sqlstore/post_acknowledgements_store.go:116]: nit: slightly weird to have this one called `ack` and elsewhere called `acknowledgement`

**Review by lieut-data — APPROVED** [review]
Nice fix, @koox00! I had to confirm for myself by comparing with the `ReactionStore` that invalidating the whole channel cache was necessary, but it does seem to be. Something to evaluate changing in the future 🤔 

A few style nits only.

**Review by vish9812 — APPROVED** [review]

**Pull request merged**